File: output/csharp/materialsystem2.cs
public static class MaterialParam_t {

   public const nint m_name = 0x0;

}

public static class MaterialParamInt_t {

   public const nint m_nValue = 0x8;

}

public static class MaterialParamFloat_t {

   public const nint m_flValue = 0x8;

}

public static class MaterialParamVector_t {

   public const nint m_value = 0x8;

}

public static class MaterialParamTexture_t {

   public const nint m_pValue = 0x8;

}

public static class MaterialParamString_t {

   public const nint m_value = 0x8;

}

public static class MaterialParamBuffer_t {

   public const nint m_value = 0x8;

}

public static class MaterialResourceData_t {

   public const nint m_materialName = 0x0;
   public const nint m_shaderName = 0x8;
   public const nint m_intParams = 0x10;
   public const nint m_floatParams = 0x28;
   public const nint m_vectorParams = 0x40;
   public const nint m_textureParams = 0x58;
   public const nint m_dynamicParams = 0x70;
   public const nint m_dynamicTextureParams = 0x88;
   public const nint m_intAttributes = 0xa0;
   public const nint m_floatAttributes = 0xb8;
   public const nint m_vectorAttributes = 0xd0;
   public const nint m_textureAttributes = 0xe8;
   public const nint m_stringAttributes = 0x100;
   public const nint m_renderAttributesUsed = 0x118;

}

public static class PostProcessingTonemapParameters_t {

   public const nint m_flExposureBias = 0x0;
   public const nint m_flShoulderStrength = 0x4;
   public const nint m_flLinearStrength = 0x8;
   public const nint m_flLinearAngle = 0xc;
   public const nint m_flToeStrength = 0x10;
   public const nint m_flToeNum = 0x14;
   public const nint m_flToeDenom = 0x18;
   public const nint m_flWhitePoint = 0x1c;
   public const nint m_flLuminanceSource = 0x20;
   public const nint m_flExposureBiasShadows = 0x24;
   public const nint m_flExposureBiasHighlights = 0x28;
   public const nint m_flMinShadowLum = 0x2c;
   public const nint m_flMaxShadowLum = 0x30;
   public const nint m_flMinHighlightLum = 0x34;
   public const nint m_flMaxHighlightLum = 0x38;

}

public static class PostProcessingBloomParameters_t {

   public const nint m_blendMode = 0x0;
   public const nint m_flBloomStrength = 0x4;
   public const nint m_flScreenBloomStrength = 0x8;
   public const nint m_flBlurBloomStrength = 0xc;
   public const nint m_flBloomThreshold = 0x10;
   public const nint m_flBloomThresholdWidth = 0x14;
   public const nint m_flSkyboxBloomStrength = 0x18;
   public const nint m_flBloomStartValue = 0x1c;
   public const nint m_flBlurWeight = 0x20;
   public const nint m_vBlurTint = 0x34;

}

public static class PostProcessingVignetteParameters_t {

   public const nint m_flVignetteStrength = 0x0;
   public const nint m_vCenter = 0x4;
   public const nint m_flRadius = 0xc;
   public const nint m_flRoundness = 0x10;
   public const nint m_flFeather = 0x14;
   public const nint m_vColorTint = 0x18;

}

public static class PostProcessingLocalContrastParameters_t {

   public const nint m_flLocalContrastStrength = 0x0;
   public const nint m_flLocalContrastEdgeStrength = 0x4;
   public const nint m_flLocalContrastVignetteStart = 0x8;
   public const nint m_flLocalContrastVignetteEnd = 0xc;
   public const nint m_flLocalContrastVignetteBlur = 0x10;

}

public static class PostProcessingResource_t {

   public const nint m_bHasTonemapParams = 0x0;
   public const nint m_toneMapParams = 0x4;
   public const nint m_bHasBloomParams = 0x40;
   public const nint m_bloomParams = 0x44;
   public const nint m_bHasVignetteParams = 0xb4;
   public const nint m_vignetteParams = 0xb8;
   public const nint m_bHasLocalContrastParams = 0xdc;
   public const nint m_localConstrastParams = 0xe0;
   public const nint m_nColorCorrectionVolumeDim = 0xf4;
   public const nint m_colorCorrectionVolumeData = 0xf8;
   public const nint m_bHasColorCorrection = 0x110;

}

File: output/py/materialsystem2.py




class MaterialParam_t:

   m_name: int = 0



class MaterialParamInt_t:

   m_nValue: int = 8



class MaterialParamFloat_t:

   m_flValue: int = 8



class MaterialParamVector_t:

   m_value: int = 8



class MaterialParamTexture_t:

   m_pValue: int = 8



class MaterialParamString_t:

   m_value: int = 8



class MaterialParamBuffer_t:

   m_value: int = 8



class MaterialResourceData_t:

   m_materialName: int = 0
   m_shaderName: int = 8
   m_intParams: int = 16
   m_floatParams: int = 40
   m_vectorParams: int = 64
   m_textureParams: int = 88
   m_dynamicParams: int = 112
   m_dynamicTextureParams: int = 136
   m_intAttributes: int = 160
   m_floatAttributes: int = 184
   m_vectorAttributes: int = 208
   m_textureAttributes: int = 232
   m_stringAttributes: int = 256
   m_renderAttributesUsed: int = 280



class PostProcessingTonemapParameters_t:

   m_flExposureBias: int = 0
   m_flShoulderStrength: int = 4
   m_flLinearStrength: int = 8
   m_flLinearAngle: int = 12
   m_flToeStrength: int = 16
   m_flToeNum: int = 20
   m_flToeDenom: int = 24
   m_flWhitePoint: int = 28
   m_flLuminanceSource: int = 32
   m_flExposureBiasShadows: int = 36
   m_flExposureBiasHighlights: int = 40
   m_flMinShadowLum: int = 44
   m_flMaxShadowLum: int = 48
   m_flMinHighlightLum: int = 52
   m_flMaxHighlightLum: int = 56



class PostProcessingBloomParameters_t:

   m_blendMode: int = 0
   m_flBloomStrength: int = 4
   m_flScreenBloomStrength: int = 8
   m_flBlurBloomStrength: int = 12
   m_flBloomThreshold: int = 16
   m_flBloomThresholdWidth: int = 20
   m_flSkyboxBloomStrength: int = 24
   m_flBloomStartValue: int = 28
   m_flBlurWeight: int = 32
   m_vBlurTint: int = 52



class PostProcessingVignetteParameters_t:

   m_flVignetteStrength: int = 0
   m_vCenter: int = 4
   m_flRadius: int = 12
   m_flRoundness: int = 16
   m_flFeather: int = 20
   m_vColorTint: int = 24



class PostProcessingLocalContrastParameters_t:

   m_flLocalContrastStrength: int = 0
   m_flLocalContrastEdgeStrength: int = 4
   m_flLocalContrastVignetteStart: int = 8
   m_flLocalContrastVignetteEnd: int = 12
   m_flLocalContrastVignetteBlur: int = 16



class PostProcessingResource_t:

   m_bHasTonemapParams: int = 0
   m_toneMapParams: int = 4
   m_bHasBloomParams: int = 64
   m_bloomParams: int = 68
   m_bHasVignetteParams: int = 180
   m_vignetteParams: int = 184
   m_bHasLocalContrastParams: int = 220
   m_localConstrastParams: int = 224
   m_nColorCorrectionVolumeDim: int = 244
   m_colorCorrectionVolumeData: int = 248
   m_bHasColorCorrection: int = 272